src/sentry/tasks/groupowner.py

```diff
@@ -64,32 +52,30 @@ def process_suspect_commits(event, **kwargs):
                     )
                     can_process = False
 
-            owner_data = {"count": owner_count, "time": timezone.now()}
-            cache.set(cache_key, owner_data, OWNER_CACHE_LIFE)
+            cache.set(cache_key, True, OWNER_CACHE_LIFE)
 
         if can_process:
             with metrics.timer("sentry.tasks.process_suspect_commits.process_loop"):
                 metrics.incr("sentry.tasks.process_suspect_commits.calculated")
                 try:
-
                     with metrics.timer(
                         "sentry.tasks.process_suspect_commits.get_serialized_event_file_committers"
                     ):
                         committers = get_event_file_committers(project, event)
-                    new_owners = []
+                    unsorted_owners = {}
                     for committer in committers:
                         if "id" in committer["author"]:
                             author_id = committer["author"]["id"]
                             for commit, score in committer["commits"]:
-                                if score >= MIN_COMMIT_SCORE and not [
-                                    aid for aid, score in new_owners if aid == author_id
-                                ]:
-                                    new_owners.append((author_id, score))
+                                if score >= MIN_COMMIT_SCORE:
+                                    unsorted_owners[author_id] = max(
+                                        score, unsorted_owners.get(author_id, 0)
+                                    )
 
-                    if new_owners:
-                        for owner_id, score in sorted(new_owners, key=lambda a: a[1], reverse=True)[
-                            :PREFERRED_GROUP_OWNERS
-                        ]:
+                    if unsorted_owners:
+                        for owner_id in sorted(
+                            unsorted_owners, reverse=True, key=unsorted_owners.get
+                        )[:PREFERRED_GROUP_OWNERS]:
```

Comment:
Probably worth adding a brief test to make sure we store the highest scoring owners

Reply:
Added a test around this